lightning/src/ln/channelmanager.rs

```diff
@@ -5251,18 +5327,25 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 		let pending_outbound_payments = self.pending_outbound_payments.lock().unwrap();
 		// For backwards compat, write the session privs and their total length.
 		let mut num_pending_outbounds_compat: u64 = 0;
-		for (_, outbounds) in pending_outbound_payments.iter() {
-			num_pending_outbounds_compat += outbounds.len() as u64;
+		for (_, outbound) in pending_outbound_payments.iter() {
+			num_pending_outbounds_compat += outbound.num_parts() as u64;
 		}
 		num_pending_outbounds_compat.write(writer)?;
-		for (_, outbounds) in pending_outbound_payments.iter() {
-			for outbound in outbounds.iter() {
-				outbound.write(writer)?;
+		for (_, outbound) in pending_outbound_payments.iter() {
+			match outbound {
+				PendingOutboundPayment::Legacy { session_privs } |
+				PendingOutboundPayment::Retryable { session_privs, .. } => {
+					for session_priv in session_privs.iter() {
+						session_priv.write(writer)?;
+					}
+				}
 			}
 		}
 
+		let pending_outbound_payments_compat_2: HashMap<PaymentId, HashSet<[u8; 32]>> = HashMap::new();
 		write_tlv_fields!(writer, {
-			(1, pending_outbound_payments, required),
+			(1, pending_outbound_payments_compat_2, required),
+			(3, pending_outbound_payments, required),
```

Comment:
Will changing the order break the format?

Reply:
If e.g. in 0.0.102 we change the hashmap serialization format that corresponds to the `1` tlv type here, 0.0.101 clients will fail to deserialize it (hope that answers)